3-Authorization-II/1-call-api/API/app.js

```diff
@@ -0,0 +1,120 @@
+const express = require('express');
+const morgan = require('morgan');
+const cors = require('cors');
+
+const passport = require('passport');
+const passportAzureAd = require('passport-azure-ad');
+
+const authConfig = require('./authConfig');
+const router = require('./routes/index');
+
+const { requiredScopeOrAppPermission } = require('./auth/permissionUtils')
+
+const app = express();
+
+/**
+ * Enable CORS middleware. In production, modify as to allow only designated origins and methods.
+ * If you are using Azure App Service, we recommend removing the line below and configure CORS on the App Service itself.
+ */
+app.use(cors());
+
+app.use(express.json());
+app.use(express.urlencoded({ extended: false }));
+app.use(morgan('dev'));
+
+const bearerStrategy = new passportAzureAd.BearerStrategy({
+    identityMetadata: `https://${authConfig.metadata.authority}/${authConfig.credentials.tenantID}/${authConfig.metadata.version}/${authConfig.metadata.discovery}`,
+    issuer: `https://${authConfig.metadata.authority}/${authConfig.credentials.tenantID}/${authConfig.metadata.version}`,
+    clientID: authConfig.credentials.clientID,
+    audience: authConfig.credentials.clientID, // audience is this application
+    validateIssuer: authConfig.settings.validateIssuer,
+    passReqToCallback: authConfig.settings.passReqToCallback,
+    loggingLevel: authConfig.settings.loggingLevel,
+    loggingNoPII: authConfig.settings.loggingNoPII,
+}, (req, token, done) => {
+    /**
+     * Below you can do extended token validation and check for additional claims, such as:
+     * - check if the caller's tenant is in the allowed tenants list via the 'tid' claim (for multi-tenant applications)
+     * - check if the caller's account is homed or guest via the 'acct' optional claim
+     * - check if the caller belongs to right roles or groups via the 'roles' or 'groups' claim, respectively
+     *
+     * Bear in mind that you can do any of the above checks within the individual routes and/or controllers as well.
+     * For more information, visit: https://docs.microsoft.com/azure/active-directory/develop/access-tokens#validate-the-user-has-permission-to-access-this-data
+     */
+
+
+    /**
+     * Below we verify if the caller's tenant ID is in the list of allowed tenants.
+     * Since this app is not configured to be multi-tenant, this is only for illustration
+     */
+    const myAllowedTenantsList = [
+        authConfig.credentials.tenantID,
+        // ...
+    ]
+
+    if (!myAllowedTenantsList.includes(authConfig.credentials.tenantID)) {
+        return done(new Error('Unauthorized'), {}, "Tenant not allowed");
+    }
```

Comment:
last minute ask, since issuer validation is tenantId validation, can we change this to verify the caller client app instead? code snippet from 4-1

            //              string clientappId = context?.Principal?.Claims
            //                  .FirstOrDefault(x => x.Type == "azp" || x.Type == "appid")?.Value;

            //              if (!allowedClientApps.Contains(clientappId))
            //              {
            //                  throw new HttpResponseException(HttpStatusCode.Unauthorized);
            //              }